tests/test_unittest/test_gtc/test_gtir.py

```diff
@@ -91,43 +89,148 @@ def test_naive_python_copy(copy_computation):
-                                                    name="b",
-                                                    offset=CartesianOffset(i=-1, j=0, k=0),
-                                                ),
-                                                right=FieldAccess(
-                                                    name="b", offset=CartesianOffset(i=1, j=0, k=0)
-                                                ),
-                                                op=BinaryOperator.ADD,
-                                            ),
-                                        )
-                                    )
-                                ],
+        vertical_loops=[
+            VerticalLoop(
+                loop_order=LoopOrder.FORWARD,
+                vertical_intervals=[
+                    VerticalInterval(
+                        start=AxisBound(level=LevelMarker.START, offset=0),
+                        end=AxisBound(level=LevelMarker.END, offset=0),
+                        body=[
+                            ParAssignStmt(
+                                left=FieldAccess.centered(name="a"),
+                                right=BinaryOp(
+                                    left=FieldAccess(
+                                        name="b",
+                                        offset=CartesianOffset(i=-1, j=0, k=0),
+                                    ),
+                                    right=FieldAccess(
+                                        name="b", offset=CartesianOffset(i=1, j=0, k=0)
+                                    ),
+                                    op=BinaryOperator.ADD,
+                                ),
                             )
                         ],
                     )
-                ]
+                ],
             )
         ],
     )
     assert ast.parse(PythonNaiveCodegen.apply(horizontal_avg))
+
+
+def test_ExprBaseclassIsNotInstantiatable():
+    with pytest.raises(TypeError):
+        Expr()
+
+
+def test_StmtBaseclassIsNotInstantiatable():
+    with pytest.raises(TypeError):
+        Stmt()
+
+
+class DummyExpr(Expr):
+    """Fake expression for cases where a concrete expression is not needed."""
+
+
+# Validation tests
+def test_ParAssignStmtWithVerticalOffsetIsOk():
+    ParAssignStmt(
+        left=FieldAccess(name="foo", offset=CartesianOffset(i=0, j=0, k=1)), right=DummyExpr()
+    )
+
+
+def test_ParAssignStmtWithHorizontalOffsetIsError():
+    with pytest.raises(ValidationError):
+        ParAssignStmt(
+            left=FieldAccess(name="foo", offset=CartesianOffset(i=1, j=0, k=0)), right=DummyExpr()
+        )
+
+
+def test_TernaryOpValidation():
+    assert (
+        TernaryOp(
+            cond=DummyExpr(dtype=DataType.BOOL),
+            true_expr=DummyExpr(dtype=DataType.INT32),
+            false_expr=DummyExpr(dtype=DataType.INT32),
+        ).dtype
+        == DataType.INT32
+    )
+
+    with pytest.raises(ValidationError):
```

Comment:
try fixture

Reply:
I decided it doesn't make sense here, because I want to have the different reasons for the failure in the test name.

Reply:
pytest injects the fixture name into the test name at runtime.  I tend to only make separate functions if the actual check is different (different exception raised for example)

Reply:
Ah, right. You said that already...

Reply:
Does it work in the case where the fixture raises the exception?

Reply:
I can return a lambda, does it make sense?